packages/service-library/src/servicelib/redis.py

```diff
@@ -205,35 +214,47 @@ async def setup(self) -> None:
         self._health_check_task = start_periodic_task(
             self._check_health,
             interval=self.health_check_interval,
-            task_name="redis_service_health_check",
+            task_name=f"redis_service_health_check_{self.redis_dsn}",
         )
 
     async def shutdown(self) -> None:
         if self._health_check_task:
-            await stop_periodic_task(self._health_check_task)
+            await stop_periodic_task(self._health_check_task, timeout=1)
         await super().shutdown()
 
 
+@dataclass(frozen=True)
+class RedisManagerDBConfig:
+    database: RedisDatabase
+    decode_responses: bool = _DEFAULT_DECODE_RESPONSES
+    health_check_interval: datetime.timedelta = _DEFAULT_HEALTH_CHECK_INTERVAL
+
+
 @dataclass
 class RedisClientsManager:
     """
     Manages the lifetime of redis client sdk connections
     """
 
-    databases: set[RedisDatabase]
+    db_configs: set[RedisManagerDBConfig]
     settings: RedisSettings
 
-    _client_sdks: dict[RedisDatabase, RedisClientSDK] = field(default_factory=dict)
+    _client_sdks: dict[RedisDatabase, RedisClientSDKHealthChecked] = field(
```

Comment:
Question: do we use the RedisClientSDK without healthcheck anywhere? if yes why would one do that? I find it quite a bit confusing now.

Reply:
It's used everywhere except for the `dynamic-sidecar`. This is the only service that actually cares about the connection to redis in its helathcheck.

It was recently introduced, though.

Maybe we should alright all other service sto use the `RedisClientSDKHealthChecked` and integrate it in their health checks?

Reply:
so why not change it in the dynamic sidecar, and change the name back to RedisClientSDK?? it would be much less confusing.

Reply:
OK, I've merged both clients.